From 3ab465f5b0e95547e31e1ed2d846c91366680695 Mon Sep 17 00:00:00 2001
From: "qinglin.li" <qinglin.li@amlogic.com>
Date: Mon, 26 Sep 2022 14:41:47 +0800
Subject: [PATCH 52/95] freertos: androidT bring up freertos with t7c [1/2]

PD#SWPL-94126

Problem:
bring up freertos with t7c

Solution:
bring up freertos with t7c

Verify:
t7c

Signed-off-by: qinglin.li <qinglin.li@amlogic.com>
Change-Id: If4661d3b705bd7af18f801e0c5131dd5492f409e
---
 arch/arm64/kernel/smp.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm64/kernel/smp.c b/arch/arm64/kernel/smp.c
index 5bdcdce62f17..764511be78e7 100644
--- a/arch/arm64/kernel/smp.c
+++ b/arch/arm64/kernel/smp.c
@@ -808,6 +808,9 @@ static const char *ipi_types[NR_IPI] __tracepoint_string = {
 	[IPI_TIMER]		= "Timer broadcast interrupts",
 	[IPI_IRQ_WORK]		= "IRQ work interrupts",
 	[IPI_WAKEUP]		= "CPU wake-up interrupts",
+#if IS_ENABLED(CONFIG_AMLOGIC_FREERTOS)
+	[IPI_FREERTOS]		= "CPU freertos interrupts",
+#endif
 };
 
 static void smp_cross_call(const struct cpumask *target, unsigned int ipinr);
@@ -854,6 +857,13 @@ void arch_irq_work_raise(void)
 }
 #endif
 
+#if IS_ENABLED(CONFIG_AMLOGIC_FREERTOS_T7)
+void arch_send_ipi_rtos(int cpu)
+{
+	smp_cross_call(cpumask_of(cpu), IPI_FREERTOS);
+}
+#endif
+
 static void local_cpu_stop(void)
 {
 	set_cpu_online(smp_processor_id(), false);
-- 
2.25.1

